baseplate/frameworks/thrift/__init__.py

```diff
@@ -83,6 +137,16 @@ def call_with_context(*args: Any, **kwargs: Any) -> Any:
                 # a normal result
                 span.finish()
                 return result
```

Comment:
if you're returning here wouldn't that prevent the `finally` block from being called?

Reply:
No, the finally block is ran before the return here.

"A finally clause is executed whether or not an exception has occurred in the try clause. When an exception has occurred, it is re-raised after the finally clause is executed. The finally clause is also executed ``on the way out'' when the try statement is left via a break or return statement. " (https://documentation.help/Python-PEP/node101.html)